lib/cocoapods/generator/info_plist_file.rb

```diff
@@ -108,15 +98,15 @@ def info
           'CFBundleInfoDictionaryVersion' => '6.0',
           'CFBundleName' => '${PRODUCT_NAME}',
           'CFBundlePackageType' => bundle_package_type.to_s.upcase,
-          'CFBundleShortVersionString' => target_version,
+          'CFBundleShortVersionString' => version,
           'CFBundleSignature' => '????',
           'CFBundleVersion' => '${CURRENT_PROJECT_VERSION}',
-          'NSPrincipalClass' => '',
           'CFBundleDevelopmentRegion' => 'en',
         }
 
         info['CFBundleExecutable'] = '${EXECUTABLE_NAME}' if bundle_package_type != :bndl
         info['CFBundleVersion'] = '1' if bundle_package_type == :bndl
+        info['NSPrincipalClass'] = 'NSApplication' if bundle_package_type == :appl && platform == :osx
```

Comment:
According to this https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW11

`NSPrincipalClass` is only present for macOS targets and must be set when generating an app host for macOS targets.

Reply:
Btw this changed because generating an app host for macOS requires this value to be set to `NSApplication` so I fixed it with this PR.